src/extras.jl

```diff
@@ -56,14 +64,76 @@ also accept them.
 * `extend::Bool=false`: when `false`, an error is raised if some values in `x` fall
   outside of the breaks; when `true`, breaks are automatically added to include all
   values in `x`, and the upper bound is included in the last interval.
-* `labels::AbstractVector=[]`: a vector of strings giving the names to use for the
-  intervals; if empty, default labels are used.
+* `labels::Union{AbstractVector,Function}: a vector of strings giving the names to use for
+  the intervals; or a function `f(from, to, i; extend)` that generates the labels from the
+  left and right interval boundaries and the group index. Defaults to 
+  `string("[", from, ", ", to, extend ? "]" : ")")`, e.g. `"[1, 5)"`.
 * `allow_missing::Bool=true`: when `true`, values outside of breaks result in missing values.
   only supported when `x` accepts missing values.
+  
+# Examples
+```jldoctest
+julia> using CategoricalArrays
```

Comment:
Is this actually needed? There's probably a way to avoid it as it takes a lot of space.

Reply:
Could you please clarify what you mean by "this"

Reply:
I mean this `using` line.

Reply:
thanks - I dropped it. Let's see what CI says